### PR TITLE
HW5 Draft Solutions

### PR DESCRIPTION
**Pull request by PSanthosh22**
This PR contains changes to the repo for HW5 solutions. 
The changes add a GitHub actions workflow that runs pylint and pytest. 
I also added a new file, CONTRIBUTING.md, for non-public-facing, developer-specific inforamation.

### CODE REVIEW

**Review by erikj540 — APPROVED** [review]

File: .github/workflows/pytest-and-pylint.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    branches:
+      - main
+
+jobs:
+  run_pylint:
+    name: Run pylint
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: bash -el {0}
```
Comment (erikj540) [inline .github/workflows/pytest-and-pylint.yml:17]: I'm curious why we need this line. Can it be removed?

File: tests/unit/test_utils.py [hunk 2]
```diff
@@ -48,9 +48,9 @@ class TestGetColmValsAsFloats:
     def test_basic():
         """Check that function works as expected on non-problematic input"""
         # Arrange
-        rows = [["1", "2"], ["4", "5"]]
+        rows = [["1", "2"], ["4", "-5.5"]]
         colm_index = 1
-        expected_output = [2, 5]
+        expected_output = [2, -5.5]
```
Comment (erikj540) [inline tests/unit/test_utils.py:53]: Why are we testing a negative number? I would think that the expected outputs would all be positive.

File: .github/workflows/pytest-and-pylint.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+name: Run pytest and pylint
```
Comment (PSanthosh22) [inline .github/workflows/pytest-and-pylint.yml:1]:
```python
print("hello")
```

**Review by erikj540 — COMMENTED** [review]
Great job 👍 

Just a few comments about things I didn't understand. Thanks!

File: .github/workflows/pytest-and-pylint.yml (hunk 1 above)
Comment (PSanthosh22) [reply .github/workflows/pytest-and-pylint.yml:17]: No, you may not

**Pull request merged**